feat: Added stat-service

rocketman27 opened the pull request:
- Added stats-dto, stats-client, stats-server,
- Added docker-compose.yml, Dockerfiles for main-service and stats-service,
- Added main classes for 2 main modules

rocketman27 committed:
feat: Added stat-service
- Added stats-dto, stats-client, stats-server,
- Added docker-compose.yml, Dockerfiles for main-service and stats-service,
- Added main classes for 2 main modules

utausheva commented on pom.xml:
```diff
@@ -22,9 +27,88 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 	</properties>
 
+	<dependencyManagement>
```
насколько я понимаю, этот блок здесь не нужен

rocketman27 replied:
Исправлено

utausheva commented on stats-service/pom.xml:
```diff
@@ -0,0 +1,50 @@
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencies>
```
и этот блок здесь тоже не нужен, зависимости описываются в каждом модуле отдельно

rocketman27 replied:
Исправлено

utausheva commented on stats-service/stats-dto/src/main/java/ru/practicum/dto/RequestInfoSummaryDto.java:
```diff
@@ -0,0 +1,12 @@
+public class RequestInfoSummaryDto {
+    private String app;
+    private String uri;
+    private Integer hits;
```
Integer точно будет достаточно?

rocketman27 replied:
Может, и нет. Изменено на Long

utausheva commented on stats-service/stats-server/src/main/java/ru/practicum/server/models/RequestInfo.java:
```diff
@@ -0,0 +1,38 @@
+package ru.practicum.server.models;
```
пакеты принято называть в единственном числе, поправь везде

rocketman27 replied:
Исправлено

utausheva commented on stats-service/stats-server/src/main/java/ru/practicum/server/models/RequestInfo.java:
```diff
@@ -0,0 +1,38 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private long id;
+    private String app;
```
также стоит перенести на модель ограничения из БД - размер length и nullable

utausheva replied:
для всех полей

rocketman27 replied:
Исправлено

utausheva reviewed (commented):
Привет!
По коду всё отлично)
Есть пара комментариев к оформлению и настройке помов)

rocketman27 committed:
fix: Refactored stat-service
- Renamed packages,
- Updated pom files for patent its submodules,
- Added persistence annotations to RequestInfo

rocketman27 commented:
> Привет!
> По коду всё отлично)
> Есть пара комментариев к оформлению и настройке помов)

Спасибо!

utausheva reviewed (approved):
Всё отлично!
